test/e2e/dataplane/service_discovery.go

```diff
@@ -42,4 +50,69 @@ func RunServiceDiscoveryTest(f *framework.Framework) {
 
 	Expect(err).NotTo(HaveOccurred())
 	Expect(stdout).To(ContainSubstring("Welcome to nginx!"))
+
+	kubeDnsService, _ := f.ClusterClients[framework.ClusterB].CoreV1().Services("kube-system").Get("kube-dns", metav1.GetOptions{})
+	kubeDnsServiceIP := kubeDnsService.Spec.ClusterIP
+
+	cmd = []string{"dig", "@" + kubeDnsServiceIP, "nginx-demo." + f.Namespace + ".svc.cluster2.local", "+short"}
+	stdout, _, err = f.ExecWithOptions(framework.ExecOptions{
+		Command:       cmd,
+		Namespace:     f.Namespace,
+		PodName:       netshootPodList.Items[0].Name,
+		ContainerName: netshootPodList.Items[0].Spec.Containers[0].Name,
+		CaptureStdout: true,
+		CaptureStderr: true,
+	}, framework.ClusterB)
+
+	Expect(err).NotTo(HaveOccurred())
+	Expect(stdout).To(ContainSubstring(nginxServiceClusterCIP))
```

Comment:
L54-68 are duplicated in `RunServiceDiscoveryLocalTest` - please extract into a function.

Reply:
Done.